### PR TITLE
front-to-back connection with user login and registration

### PR DESCRIPTION
**Pull request by Fatimata-dev**
creation of a user service, implementation of methods to be able to register or connect a user by calling the back-end then implementation of these methods in the components (signup, login and profile)

### CODE REVIEW

**Review by MoubarkiS — APPROVED** [review]
Clean code

File: src/app/app.module.ts
Comment (Benja-san) [inline src/app/app.module.ts:1]: Do you really need this app module file ?

**Review by Benja-san — CHANGES_REQUESTED** [review]
Consider registering the user inside local storage or at least inside a subject available all along navigation. Or else you'll loose the user as soon as you quit your profile page

**Comment by Fatimata-dev** [comment]
Saves the information of the logged in user in localstorage then displays links in the nav-bar depending on the state of the user logged in or not

**Review by Benja-san — APPROVED** [review]
Don't forget to finish this us with the update and debug (I'll probably get time this afternoon or Monday)

**Pull request merged**